overlord/snapshotstate/backend/helpers.go

```diff
@@ -0,0 +1,90 @@
+package backend
+
+import (
+	"archive/zip"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"regexp"
+	"strings"
+
+	"github.com/snapcore/snapd/client"
+	"github.com/snapcore/snapd/dirs"
+	"github.com/snapcore/snapd/osutil"
+)
+
+func nextBackup(fn string) (string, error) {
+	// this is a TOCTOU problem
```

Comment:
I'm curious and worried here.

What are we racing with? Are we racing with someone other than snapd?

Reply:
if somebody spams files into there we need to worry about whether they can get us to write to a symlink, for exmaple.